### PR TITLE
Autoscaler test

### PR DESCRIPTION
**Pull request by enxebre**
Since we currently have a multirepo approach we want to validate e2e features in every repo consuming the cluster-api, i.e cluster autoscaler operator, cluster autoscaler, cluster-api-propvider-aws, machine API operator. Once the cluster autoscaler is updated to support machineDeployments this test will be skipped and will need to be updated to target machineDeployments instead

### CODE REVIEW

File: test/e2e/operator_expectations.go [hunk 1]
```diff
@@ -340,3 +345,195 @@ MachineLoop:
...
+			glog.Errorf("error querying api for nodeList object: %v, retrying...", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+	clusterInitialTotalNodes := len(nodeList.Items)
+	glog.Infof("Cluster initial numner of nodes is %d", clusterInitialTotalNodes)
```
Comment (frobware) [inline test/e2e/operator_expectations.go]: typo: "number"

File: test/e2e/operator_expectations.go [hunk 2]
```diff
@@ -340,3 +345,195 @@ MachineLoop:
...
+	})
+
+	glog.Info("Delete workload")
+	return wait.PollImmediate(1*time.Second, waitMedium, func() (bool, error) {
+		if err := tc.client.Delete(context.TODO(), &workLoad); err != nil {
+			glog.Errorf("error querying api for workLoad object: %v, retrying...", err)
+			return false, nil
+		}
+		return true, nil
+	})
```
[the remark this answers is not in the document]
Comment (ingvagabund) [reply test/e2e/operator_expectations.go]: Please remove both `clusterAutoscaler` and `machineAutoscaler` objects as well.

[the remark this answers is not in the document]
Comment (enxebre) [reply test/e2e/operator_expectations.go]: I'm deleting autoscaler object, setting back initial number of replicas and waiting for nodes to match initial number now
https://github.com/openshift/machine-api-operator/pull/177#discussion_r247471995 yes, this in intended to be an initial smoke test,  any farther improvements will be added in follow ups

**Pull request merged**